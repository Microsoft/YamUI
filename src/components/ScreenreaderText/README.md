# ScreenreaderText

This component simply hides its content from sight without hiding it from screenreaders. Use this component whenever a screenreader should be able to read aloud additional context for your UI features.

## Properties

(none)

## Notes for use

Generally just nest text within a ScreenreaderText component. It's also possible to nest JSX, for instance if a screenreader should recognize some text as the page title you could nest an `<h1>` tag.

---

## Examples

### Basic usage

```js
<ScreenreaderText>
  This text is not visible, but will still be read aloud by a screenreader.
</ScreenreaderText>.
```
