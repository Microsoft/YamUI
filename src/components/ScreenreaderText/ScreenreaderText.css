/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
.y-screenreaderText {
  display: block;
  padding: 0 !important;
  border: 0 !important;
  clip: rect(0 0 0 0) !important;
  position: absolute !important;
  margin: -1px !important;
  height: 1px !important;
  width: 1px !important;
  overflow: hidden !important;
  word-wrap: normal !important;
}
