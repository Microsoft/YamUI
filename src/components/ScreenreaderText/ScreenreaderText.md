This component simply hides its content from sight without hiding it from screenreaders. Use this component whenever a screenreader should be able to read aloud additional context for your UI features.

### Notes for use

Generally just nest text within a `ScreenreaderText` component. It's also possible to nest React nodes, for instance if a screenreader should recognize some text as the page title you could nest an `h1` tag.

### Examples

Basic usage:

```js { "props": { "data-description": "basic" } }
<div>
  There is a <strong>ScreenreaderText</strong> component with text after this visible text which is hidden in the browser but still available to screenreaders. <ScreenreaderText>You can't see this but screenreaders can.</ScreenreaderText>
</div>
```
