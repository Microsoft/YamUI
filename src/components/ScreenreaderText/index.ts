/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export { default } from './ScreenreaderText';
export * from './ScreenreaderText';
