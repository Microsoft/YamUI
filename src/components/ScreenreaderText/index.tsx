/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import classNames = require('classnames');
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import './screenreader-text.css';

export interface ScreenreaderTextProps extends NestableBaseComponentProps {}

const ScreenreaderText: React.StatelessComponent<ScreenreaderTextProps> = props => (
  <span className={classNames('y-screenreaderText', props.className)}>{props.children}</span>
);

export default ScreenreaderText;
