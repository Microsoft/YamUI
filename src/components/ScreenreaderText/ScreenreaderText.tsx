/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import classNames = require('classnames');
import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
import './ScreenreaderText.css';

export interface ScreenreaderTextProps extends NestableBaseComponentProps {}

export default class ScreenreaderText extends React.PureComponent<ScreenreaderTextProps, {}> {
  render() {
    const { className, children } = this.props;

    return <span className={classNames('y-screenreaderText', className)}>{children}</span>;
  }
}
