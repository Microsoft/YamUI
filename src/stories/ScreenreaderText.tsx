/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import ScreenreaderText from '../components/ScreenreaderText';
import withReadme from 'storybook-readme/with-readme';
const readme = require('../components/ScreenreaderText/README');

storiesOf('ScreenreaderText', module)
  .addDecorator(withReadme(readme))
  .add('default', () => (
    <div>
      There is a <strong>ScreenreaderText</strong> component with text after this visible text which
      is hidden in the browser but still available to screenreaders.
      <ScreenreaderText>You can't see this but screenreaders can</ScreenreaderText>
    </div>
  ));
