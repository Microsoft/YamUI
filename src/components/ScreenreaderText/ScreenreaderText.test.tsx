/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ScreenreaderText, { ScreenreaderTextProps } from '.';

describe('<ScreenreaderText />', () => {
  let component: ShallowWrapper<ScreenreaderTextProps>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<ScreenreaderText>text content</ScreenreaderText>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<ScreenreaderText className="TEST_CLASSNAME">test content</ScreenreaderText>);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
