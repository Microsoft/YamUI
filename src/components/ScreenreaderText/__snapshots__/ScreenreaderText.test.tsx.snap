// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenreaderText /> with additional className matches its snapshot 1`] = `
<span
  className="y-screenreaderText TEST_CLASSNAME"
>
  test content
</span>
`;

exports[`<ScreenreaderText /> with default options matches its snapshot 1`] = `
<span
  className="y-screenreaderText"
>
  text content
</span>
`;
